src/cache/redis.rs

```diff
@@ -31,8 +32,12 @@ pub struct RedisCache {
 impl RedisCache {
     /// Create a new `RedisCache`.
     pub fn new(url: &str) -> Result<RedisCache> {
+        let mut parsed = Url::parse(url)?;
+        if parsed.password().is_some() {
+            let _ = parsed.set_password(Some("*****"));
```

Comment:
nit: I would prefer a wrapper type, that has an alternative `Display` implementation rather than this hack. At the very least this requires a comment to why this is done and a statement of the restrictions following from it.

Reply:
Hack? I would consider this the simplest possible implementation for masking the password for display.

The `url` member of `RedisCache` is used only for display purposed. How about I rename this member to `displayUrl` and add a comment?

Reply:
My apologies, "hack" was definitely the wrong choice of words. There is just the possibility of this being re-used in the future and not being explicit is a potential future pain point.

Are name to `displayUrl` surely conveys this.

Reply:
No worries.

Comment added.